api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,33 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  try {
+    const token = req.header('X-Access-Token');
+    if (!token) {
+      throw new ClientError(401, 'invalid access token');
+    }
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
You don't need a `try` `catch` here, an error thrown here will already be given to the error middleware which has a check for an error from `jwt`.

As this is set up now if there was an error it would go to the `catch` then nothing would happen, the server would just get stuck. If you did use the `catch` (as it is now) you would need to call `next` in there so the server knows what to do next. But again the `try` `catch` isn't needed here.